src/Param.cc

```diff
@@ -65,54 +65,66 @@ Param::Param(const std::string &_key, const std::string &_typeName,
              const std::string &_description)
   : dataPtr(new ParamPrivate)
 {
-  this->dataPtr->key = _key;
-  this->dataPtr->required = _required;
-  this->dataPtr->typeName = _typeName;
-  this->dataPtr->description = _description;
-  this->dataPtr->set = false;
-  this->dataPtr->ignoreParentAttributes = false;
-  this->dataPtr->defaultStrValue = _default;
+  sdf::Errors errors;
+  this->dataPtr->Init(_key, _typeName, _default, _required,
+                      errors, _description);
 
-  SDF_ASSERT(
-      this->dataPtr->ValueFromStringImpl(
-          this->dataPtr->typeName,
-          _default,
-          this->dataPtr->defaultValue),
-      "Invalid parameter");
-  this->dataPtr->value = this->dataPtr->defaultValue;
-  this->dataPtr->strValue = std::nullopt;
+  if(!errors.empty())
+  {
+    for (unsigned int i = 0; i < errors.size() - 1; ++i)
+    {
+      sdferr << errors[i].Message() + "\n";
+    }
```

Comment:
This for-loop can be shortened to `sdferr << errors << "\n";`

Reply:
I think you skipped this one.

Reply:
I wanted to keep previous behavior and print all messages except the last one. Then use the last one in the `SDF_ASSERT`. I guess I could create a "subvector" and pass that to `sdferr` and then call `SDF_ASSERT` with the last message, do think that's worth it?

Reply:
Ah, okay. I'm good with that. The nice thing about doing `sdferr << errors` is that it prints other parts of the error, such as the error code and the line number (if there is one associated with the error). I won't block on this.